airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/commands.py

```diff
@@ -2,23 +2,71 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
 
+import logging
+
 import asyncclick as click
-from pipelines.airbyte_ci.test.pipeline import run_test
+from pipelines.cli.click_decorators import click_ignore_unused_kwargs
+from pipelines.consts import DOCKER_VERSION
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.contexts.click_pipeline_context import ClickPipelineContext, pass_pipeline_context
 
 
 @click.command()
 @click.argument("poetry_package_path")
 @click.option("--test-directory", default="tests", help="The directory containing the tests to run.")
+@pass_pipeline_context
+@click_ignore_unused_kwargs
 async def test(
+    ctx: ClickPipelineContext,
     poetry_package_path: str,
     test_directory: str,
 ):
     """Runs the tests for the given airbyte-ci package.
 
     Args:
+        ctx (ClickPipelineContext): The context object.
```

Comment:
I'm not sure if it matters or if it is easy to change. One thing that has been weird for me is the fact that we name both the `click.Context` and `ClickPipelineContext` `ctx` as params. The type hints help, but I still catch myself trying to use one's method for the other. 

In addition, there have been places where I want to use both (To be seen whether my use case for doing this is actually at all legit, but...) To accomplish this I have 
```
async def fix(ctx: click.Context, pipeline_ctx: ClickPipelineContext):
```
where fix is a click group... I did not think this would work at all and yet it did. Just some food for thought

Reply:
Hmm 🤔 

So I see what you mean on the naming.

Im not sure a good general path, but we could update the naming of the variable to be `pipeline_context` and just socially enforce it?

In the case of using the `ClickPipelineContext` and the `ClickContext`

You could/should use the internal ref to the `clickcontext` 
```py
async def fix(pipeline_ctx: ClickPipelineContext):
    click_context = pipeline_ctx._click_context
```

Reply:
> You could/should use the internal ref to the clickcontext
ooh, i'll try this 

> Im not sure a good general path, but we could update the naming of the variable to be pipeline_context and just socially enforce it? 
I think the below looks nice, I would be into it! 
```
async def fix(pipeline_ctx: ClickPipelineContext):
```